### PR TITLE
PETE Stack

### PR DESCRIPTION
**Pull request by nelsonic**
So... the _reason_ this PR has been  _exceptionally_ "***long time in the making***"
is because I wanted to have a _decent_ `learn-...` tutorial `forEach` element in the stack! `before` updating this doc!
Thanks to the _amazing_ people in the @dwyl community contributing to the tutorials, 
we can _finally_ update the Tech Stack document to reflect our plans for 2017...!

This Pull Request "fixes https://github.com/dwyl/technology-stack/issues/37 by adding:
+ New Diagram
+ Description of each element in the stack with links to where people can read more
+ Reasoning for considering a new stack when we are already _good_ with the "old" one!
+ Deployment details for Azure "Cloud" for HealthLocker 

Sorry that it's a bit "long"...
If I had "_more time_" I would cut it down ...
Hopefully someone `else` can provide a bit of editorial to streamline my ramblings.
Thanks for reviewing! happy to make any changes necessary.
(_but would prefer this to be merged in soon so that I can show it to Kumar+Katy ... 
and then we can iterate/improve laters... e.g. [PETE -> TEPPE](https://github.com/dwyl/technology-stack/issues/37#issuecomment-280184439) ?_)

Thanks!! ❤️

### CODE REVIEW

**Comment by nelsonic** [comment]
@jruts thanks for reading the draft and spotting the typo. I've corrected it. 👍 
@iteles, @katbow @SimonLab, @shouston3 ... please read if you have time ...
anyone can _merge_ because we can/will iterate on this!

File: README.md [hunk 1]
```diff
@@ -1,13 +1,299 @@
...
+LIFE Magazine (2 May 1955) p. 64”
+
+In November 2016 we (_once again_) **questioned our _assumptions_**,
+***re-examined*** and
+[***surveyed***](https://github.com/dwyl/learn-elm/issues/10)
+the "landscape" of "_emerging trends_" in web app development.
+We were ~~pleasantly surprised~~ ***delighted*** to see the _amazing progress_
+made by the people in the Elixir, Phoenix and Elm communities!
+These technologies are set to "***take off***" in 2017 and we are _excited_
+to be _sharing_ the technical/competitve advantage with our clients!
```
Comment (jruts) [inline README.md]: competitve > competitive

File: README.md [hunk 2]
```diff
@@ -1,13 +1,299 @@
...
+
+## What?
+
+This document + diagram _describes_ the full "**PETE**" Technology Stack
+we use for dwyl products and services (clients).
+
+Each element in our stack was _carefully_ selected based
+on it's individual merits. <br />
+When _assembled_ into a seamless "machine",
+the stack _unrivaled_ for developer productivity and word-class quality!
```
Comment (jruts) [inline README.md]: Is it supposed to be world-class quality? And if it is word-class, how do I interpret this sentence?

Comment (nelsonic) [inline README.md]: thanks for spotting the typo! just committed a fix. 👍

File: README.md [hunk 3]
```diff
@@ -1,13 +1,299 @@
...
+who worked on/with Rails. It _solves_
+all of the speed/socket/scaling/concurrency, issues
+people felt when building/using Rails apps.
+The list of ***benefits*** Phoenix has over
+(_vertually every_) other Web Frameworks is _extensive_.<br />
+Please see:
+[github.com/dwyl/learn-phoenix-framework#our-**top-10-reasons**-why-phoenix](https://github.com/dwyl/learn-phoenix-framework#our-top-10-reasons-why-phoenix) <br />
+
++ **Elixir** is the _functional_ programming language used to
+build using the Phoenix framework. Elixir is a _beautiful_ language
```
Comment (jruts) [inline README.md]: programming language used to build ~~using~~ the Phoenix framework

File: README.md [hunk 4]
```diff
@@ -1,13 +1,299 @@
...
+written _from scratch_ to be ***friendly, concise and efficient***.
+***Yes***, Elixir not a "_popular_" as JavaScript, Java, C# or PHP,
```
Comment (jruts) [inline README.md]: not **as** popular

File: README.md [hunk 5]
```diff
@@ -1,13 +1,299 @@
...
+but its' adoption is growing rapidly and most importantly its'
```
Comment (jruts) [inline README.md]: its' -> it's (2 times)

File: README.md [hunk 6]
```diff
@@ -1,13 +1,299 @@
...
+what many _experienced_ developers are gravitating towards and
+described as their ["most wanted"]()
+Also a language's popularity has more
+to do with the intellectual inertia people/companies have because
+they allow existing (_legacy_) codebases to dictate future development;
+i.e.
+[***sunk cost bias***](http://www.investopedia.com/terms/s/sunk-cost-trap.asp).
+see: [github.com/dwyl/learn-elixir#key-advantages](https://github.com/dwyl/learn-elixir#**key-advantages**)
+
++ **Tachyons** is the most _sane_ way of creating a _beatiful_ web app UI
```
Comment (jruts) [inline README.md]: _beatiful_ -> beautiful

**Comment by sohilpandya** [comment]
@jruts you are on 🔥🎉☺️

File: README.md [hunk 7]
```diff
@@ -1,13 +1,299 @@
...
+"_cascading_" of styles, Tachyons makes the style of each component _specific_
+and _local_ to that component.
+see: https://github.com/dwyl/learn-tachyons
+
++ **Elm** is a front-end developer's _dream_
+made _reality_ by a community of unbelievably
+brilliant and dedicated developers. <br />
+While Elm is
+[***incredibly fast***](http://elm-lang.org/blog/blazing-fast-html-round-two)
+the _reason_ we love elm is _nothing_ to do
```
Comment (jruts) [inline README.md]: is nothing -> has nothing

File: README.md (hunk 5 above)
Comment (nelsonic) [inline README.md]: I never know how to write `it's` or `its'` ... 
would be stoked for anyone with _much_ betterererer engrish to fix the typos in the next PR. 👍

**Comment by iteles** [comment]
@jruts Sorry I merged this out from under you, got a little carried away! Let's PR in the typo corrections :+1:

File: README.md [hunk 8]
```diff
@@ -1,13 +1,299 @@
...
+The Phoenix Application Server is hosted on (_a minimum of_)
+Two Linux Servers <br />
+(_often many more which "**message**"
+one another to distribute load as a cluster_). <br />
+The "_cluster_" is managed by Erlang's "Supervisor".
+The Erlang Supervisor is the "_Gold Standard_" in infrastructure management,
+having been used by Telecoms companies for over 20 years in production
+with some Telcos reporting 99.99999% ("_seven nines_") of "_up-time_".
+
+> It's _far_ more likely that the _infrastructure_ provider (_e.g. AWS/Azuer_)
```
Comment (jruts) [inline README.md]: AWS/Azuer > AWS/Azure

File: README.md [hunk 9]
```diff
@@ -70,11 +359,187 @@ https://github.com/dwyl/learn-istanbul#tracking-coverage-as-a-service
...
+In **1996** Nokia introduced the ["***Communicator***"](https://en.wikipedia.org/wiki/Nokia_Communicator)
+and was a **_incredible_ revolutionary innovation**!
+**Internet, Email and _Fax_** in your ***Pocket***!! <br />
+
+![nokia-communicator](https://cloud.githubusercontent.com/assets/194400/23014351/89c07b64-f426-11e6-8c5a-f93b71a3aa53.jpg) <br />
+
+Nokia continued to _dominate_ the mobile phone industry/market for the next
+***decade*** producing the _best-selling_
+**5110** and **3310** we all remember! <br />
+But by being "_ahead_" Nokia were _unable_ to see the "_contender_"
```
Comment (jruts) [inline README.md]: Nokia were > Nokia was?

File: README.md (hunk 8 above)
Comment (nelsonic) [inline README.md]: well-spotted. fixing now. 👍

File: README.md [hunk 10]
```diff
@@ -70,11 +359,187 @@ https://github.com/dwyl/learn-istanbul#tracking-coverage-as-a-service
...
+coming "_out of nowhere_" to challenge their position.
+
+In 2006 _nobody_ was making/buying "smart" mobile phones
+with glass touch screens that ran "apps" ... <br />
+in [January 2007 Steve Jobs introduced the iPhone](https://www.youtube.com/results?search_query=Steve+Jobs+iPhone+Introduction+2007)
+and _litterally_ changed the industry!
```
Comment (jruts) [inline README.md]: _litterally_ > _literally_

File: README.md (hunk 9 above)
Comment (nelsonic) [inline README.md]: agreed. fixing. 👍

File: README.md (hunk 10 above)
Comment (nelsonic) [inline README.md]: fixed.

File: README.md [hunk 11]
```diff
@@ -70,11 +359,187 @@ https://github.com/dwyl/learn-istanbul#tracking-coverage-as-a-service
...
+issues**](https://twitter.com/ThePracticalDev/status/800752571497545729)
+not **real** ones. <br />
+discussing "scalability" before you have 10,000 paying customers is
+a waste of time!!_ <br />
+
+Stop worrying about "scalability"
+and instead **focus** on building something **useful** <br />
+**focus** on **User Experience** not "backend" **scalability**!
+
+The _good_ news is that Phoenix was _does_ "***scale***" _really well_! <br />
```
Comment (jruts) [inline README.md]: Phoenix was _does_ "***scale***" > Phoenix ~~was~~ _does_ "***scale***"

File: README.md [hunk 12]
```diff
@@ -70,11 +359,187 @@ https://github.com/dwyl/learn-istanbul#tracking-coverage-as-a-service
...
+a website dedicated to the "Callback Hell" issue: http://callbackhell.com <br />
+why do we even _need_ to have an _expression_ for that?
+why can't ***elegant*** asynchronous control flow
+be the **_only_ way** to write code?
+
+
+The **fact** is: we _only_ use JavaScript because it is the
+["***Lingua Franca***"](https://en.wikipedia.org/wiki/Lingua_franca)
+that _all_ web browsers "_understand_".<br />
+It's _definately_ not because it's a "_better_" language than Python or Lisp;
```
Comment (jruts) [inline README.md]: _definately_ > _definitely_

**Comment by jruts** [comment]
@nelsonic that is all I found in my read-through.

Good stuff  <3

**Comment by nelsonic** [comment]
@jruts thanks a _million_ for _making_ time to review. ❤️ 
I've addressed the typos & grammar issues in: #46 
please check & merge if ok. thanks!! 🖖

File: README.md (hunk 5 above)
Comment (rjmk) [inline README.md]: It should be `its` in the first case and `it's` in the second case

Comment (rjmk) [inline README.md]: @SimonLab beat me to it

Comment (nelsonic) [inline README.md]: 😉

**Pull request merged**